launchers/junit/src/testFixtures/java/org/eclipse/dataspaceconnector/junit/launcher/EdcRuntimeExtension.java

```diff
@@ -0,0 +1,175 @@
+
+        var latch = new CountDownLatch(1);
+
+        runtimeThread = new Thread(() -> {
+            try {
+
+                // Make the ClassLoader available to the ServiceLoader.
+                // This ensures the target module's extensions are discovered and loaded at runtime boot.
+                Thread.currentThread().setContextClassLoader(classLoader);
+
+                // Boot EDC runtime.
+                super.beforeTestExecution(extensionContext);
+
+                latch.countDown();
+            } catch (Exception e) {
+                throw new EdcException(e);
+            }
+        });
+
+        // Start thread and wait for EDC to start up.
+        runtimeThread.start();
+
+        if (!latch.await(10, SECONDS)) {
+            throw new EdcException("Failed to start EDC runtime");
+        }
+
+        // Restore system properties.
+        System.setProperties(savedProperties);
+    }
+
+
+    @Override
+    public void afterTestExecution(ExtensionContext context) throws Exception {
+        runtimeThread.join();
+        super.afterTestExecution(context);
+    }
+
+    /**
+     * Utility method to locate the Gradle project root.
+     *
+     * @param path directory in which to start ascending search for the Gradle root.
+     * @return The Gradle project root directly, or <code>null</code> if not found.
+     */
+    private File findRoot(File path) {
+        File gradlew = new File(path, GRADLE_WRAPPER);
+        if (gradlew.exists()) {
+            return path;
+        }
+        var parent = path.getParentFile();
+        if (parent != null) {
+            return findRoot(parent);
+        }
+        return null;
+    }
+
+    /**
+     * Replace Gradle subproject JAR entries with subproject build directories in classpath.
+     * This ensures modified classes are picked up without needing to rebuild dependent JARs.
+     *
+     * @param root           project root directory.
+     * @param classPathEntry class path entry to resolve.
+     * @return resolved class path entries for the input argument.
+     */
+    private static Stream<URL> resolveClassPathEntry(File root, String classPathEntry) {
+        try {
+            File f = new File(classPathEntry).getCanonicalFile();
+
+            // If class path entry is not a JAR under the root (i.e. a sub-project), do not transform it
+            boolean isUnderRoot = f.getCanonicalPath().startsWith(root.getCanonicalPath() + File.separator);
+            if (!classPathEntry.toLowerCase(Locale.ROOT).endsWith(".jar") || !isUnderRoot) {
+                return Stream.of(new File(classPathEntry).toURI().toURL());
+            }
+
+            // Replace JAR entry with the resolved classes and resources folder
+            var buildDir = f.getParentFile().getParentFile();
+            return Stream.of(
+                    new File(buildDir, "/classes/java/main").toURI().toURL(),
+                    new File(buildDir, "/resources/main/").toURI().toURL()
```

Comment:
```suggestion
                    new File(buildDir, "/resources/main").toURI().toURL()
```

NB toURL adds the needed trailing slash